src/energyplus/ForwardTranslator/ForwardTranslateHeatExchangerDesiccantBalancedFlow.cpp

```diff
@@ -0,0 +1,127 @@
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "../ForwardTranslator.hpp"
+#include "../../model/Model.hpp"
+
+#include "../../model/HeatExchangerDesiccantBalancedFlow.hpp"
+
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../model/Schedule.hpp"
+#include "../../model/Schedule_Impl.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlowPerformanceDataType1.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlowPerformanceDataType1_Impl.hpp"
+
+#include <utilities/idd/HeatExchanger_Desiccant_BalancedFlow_FieldEnums.hxx>
+// #include "../../utilities/idd/IddEnums.hpp"
+#include <utilities/idd/IddEnums.hxx>
+
+using namespace openstudio::model;
+
+namespace openstudio {
+
+namespace energyplus {
+
+  boost::optional<IdfObject> ForwardTranslator::translateHeatExchangerDesiccantBalancedFlow(HeatExchangerDesiccantBalancedFlow& modelObject) {
+    boost::optional<std::string> s;
+    boost::optional<double> value;
+    OptionalModelObject temp;
+
+    // HeatExchangerPerformance, is required, so start by that
+    HeatExchangerDesiccantBalancedFlowPerformanceDataType1 performance = modelObject.heatExchangerPerformance();
+    boost::optional<IdfObject> _performance = translateAndMapModelObject(performance);
+    if (!_performance) {
+      LOG(Warn, modelObject.briefDescription() << " cannot be translated as its performance object cannot be translated: "
+                                               << performance.briefDescription() << ".");
+      return boost::none;
+    }
+
+    IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::HeatExchanger_Desiccant_BalancedFlow, modelObject);
+
+    // AvailabilityScheduleName
+    Schedule sched = modelObject.availabilitySchedule();
+    translateAndMapModelObject(sched);
+    idfObject.setString(HeatExchanger_Desiccant_BalancedFlowFields::AvailabilityScheduleName, sched.name().get());
+
+    // RegenerationAirInletNodeName
+    temp = modelObject.primaryAirInletModelObject();
+    if (temp) {
+      s = temp->name();
+      if (s) {
+        idfObject.setString(HeatExchanger_Desiccant_BalancedFlowFields::RegenerationAirInletNodeName, *s);
+      }
```

Comment:
I dislike this construct, but maybe I'm just pedantic. `setString(..., temp->nameString())` is shorter without an extra check (it's a node, we *know* it has a nameString)

Reply:
I was just following `HeatExchangerAirToAirSensibleAndLatent`.